clients/include/testing_axpy_batched.hpp

```diff
@@ -44,94 +47,136 @@ hipblasStatus_t testing_axpy_batched(const Arguments& argus)
     int sizeY = N * abs_incy;
     T   alpha = argus.alpha;
 
-    double gpu_time_used, cpu_time_used;
-    double rocblas_error = 0.0;
+    double gpu_time_used, hipblas_error_host, hipblas_error_device;
 
     hipblasHandle_t handle;
     hipblasCreate(&handle);
 
     // Naming: dX is in GPU (device) memory. hK is in CPU (host) memory, plz follow this practice
-    host_vector<T> hx_array[batch_count];
-    host_vector<T> hy_array[batch_count];
-    host_vector<T> hx_cpu_array[batch_count];
-    host_vector<T> hy_cpu_array[batch_count];
-
-    device_batch_vector<T> bx_array(batch_count, sizeX);
-    device_batch_vector<T> by_array(batch_count, sizeY);
+    host_batch_vector<T> hx(N, incx, batch_count);
+    host_batch_vector<T> hy_host(N, incy, batch_count);
+    host_batch_vector<T> hy_device(N, incy, batch_count);
+    host_batch_vector<T> hx_cpu(N, incx, batch_count);
+    host_batch_vector<T> hy_cpu(N, incy, batch_count);
+
+    device_batch_vector<T> dx(N, incx, batch_count);
+    device_batch_vector<T> dy_host(N, incy, batch_count);
+    device_batch_vector<T> dy_device(N, incy, batch_count);
+    device_vector<T>       d_alpha(1);
+    CHECK_HIP_ERROR(dx.memcheck());
+    CHECK_HIP_ERROR(dy_host.memcheck());
+    CHECK_HIP_ERROR(dy_device.memcheck());
+
+    hipblas_init(hx, true);
+    hipblas_init(hy_host, false);
+    hy_device.copy_from(hy_host);
+    hx_cpu.copy_from(hx);
+    hy_cpu.copy_from(hy_host);
+
+    CHECK_HIP_ERROR(dx.transfer_from(hx));
+    CHECK_HIP_ERROR(dy_host.transfer_from(hy_host));
+    CHECK_HIP_ERROR(dy_device.transfer_from(hy_device));
+    CHECK_HIP_ERROR(hipMemcpy(d_alpha, &alpha, sizeof(T), hipMemcpyHostToDevice));
+    /* =====================================================================
+         HIPBLAS
+    =================================================================== */
+    status_1 = hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_DEVICE);
```

Comment:
If you make the localHandle destructor you can also switch to use the macro for each of these to be CHECK_HIPBLAS and return on the first failure.   I'll shut up now.